Core/UserAgentManager.swift

```diff
@@ -23,16 +23,20 @@ import WebKit
 public class UserAgentManager {
     
     public static let shared = UserAgentManager()
-    
-    private var defaultAgentRetreived = false
-    private var userAgent: UserAgent
+
+    private var userAgent = UserAgent()
     
     init() {
+        DispatchQueue.main.async {
```

Comment:
This leads to a deadlock that then freezes browsing until the timeout of 2 seconds. I like the approach overall though and since we're using the fallback UA until we can get the real one, we could make the js code async. I have hacked out an example here https://github.com/duckduckgo/iOS/compare/develop...feature/mia/user-agent-crash-fix. What do you think?

Reply:
Oh yes, just using completion handler is much better 🤦‍♂️ 

I've tweaked my PR (Github has some issues - had to change the base branch in order to make it refresh this...) to match your idea.

Thanks! :)